src/hexkit/opentelemetry_setup.py

```diff
@@ -61,3 +66,105 @@ def traced_function(*args, **kwargs):
             return traced_function
 
         return outer_wrapper
+
+
+class OpenTelemetryConfig(BaseSettings):
+    """OpenTelemetry specific configuration options"""
+
+    enable_opentelemetry: bool = Field(
+        default=False,
+        description="If set to true, this will run necessary setup code."
+        "If set to false, environment variables are set that should also effectively "
+        "disable autoinstrumentation.",
+    )
+    samplig_rate: Annotated[float, Field(strict=True, ge=0, le=1)] = Field(
+        default=1.0,
+        description="Determines which proportion of spans should be sampled. "
+        "A value of 1.0 means all and is equivalent to the previous behaviour. "
+        "Setting this to 0 will result in no spans being sampled, but this does not "
+        "automatically set `enable_opentelemetry` to False.",
+    )
+    protocol: Literal["grpc", "http/protobuf"] = Field(
+        default="http/protobuf",
+        description="Specifies which protocol should be used by exporters.",
+    )
+
+
+def configure_opentelemetry(*, service_name: str, config: OpenTelemetryConfig):
+    """Configure all needed parts of OpenTelemetry.
+
+    Setup of the TracerProvider is done programmatically, all other configuration exports
+    OpenTelemetry specific environment variables.
+    """
+    configure_tracer(service_name=service_name, config=config)
+
+
+@deprecated(
+    "This function will be removed in v5. Use `configure_opentelemetry instead.`"
+)
+def configure_tracer(*, service_name: str, config: OpenTelemetryConfig):
+    """Configure all needed parts of OpenTelemetry.
+
+    Setup of the TracerProvider is done programmatically, all other configuration exports
+    OpenTelemetry specific environment variables.
+    """
+    global TRACER
+    # opentelemetry distro sets this to grpc, but in the current context http/protobuf is preferred
+    os.environ.setdefault(OTEL_EXPORTER_OTLP_PROTOCOL, config.protocol)
+    # Disable OpenTelemetry metrics and logs explicitly as they are not processed in the backend currently
+    # This overwrites the defaults of `otlp` set in opentelemetry distro
+    os.environ.setdefault(OTEL_METRICS_EXPORTER, "none")
```

Comment:
Actually, if the distro sets `OTEL_METRICS_EXPORTER="otlp"` then the `.setdefault` will not change it. Is this what we want? Or do we rather want to unconditionally set `os.environ[OTEL_METRICS_EXPORTER] = "none"`?

Reply:
Thanks for catching that, copied that from a testing context and didn't think about it not overwriting what's already set. I'll adjust that